tests/test_ice_servers.py

```diff
@@ -0,0 +1,162 @@
+"""Test the ICE servers module."""
+
+import asyncio
+import time
+
+import pytest
+from webrtc_models import RTCIceServer
+
+from hass_nabucasa import ice_servers
+
+
+@pytest.fixture
+def ice_servers_api(auth_cloud_mock) -> ice_servers.IceServers:
+    """ICE servers API fixture."""
+    auth_cloud_mock.servicehandlers_server = "example.com/test"
+    auth_cloud_mock.id_token = "mock-id-token"
+    return ice_servers.IceServers(auth_cloud_mock)
+
+
+@pytest.fixture(autouse=True)
+def mock_ice_servers(aioclient_mock):
+    """Mock ICE servers."""
+    aioclient_mock.get(
+        "https://example.com/test/webrtc/ice_servers",
+        json=[
+            {
+                "urls": "turn:example.com:80",
+                "username": "12345678:test-user",
+                "credential": "secret-value",
+            },
+        ],
+    )
+
+
+async def test_ice_servers_listener_registration_triggers_periodic_ice_servers_update(
+    ice_servers_api: ice_servers.IceServers,
+):
+    """Test that registering an ICE servers listener triggers a periodic update."""
+    times_register_called_successfully = 0
+
+    ice_servers_api._get_refresh_sleep_time = lambda: -1
```

Comment:
This can't happen anymore, right? We have the guard now that returns the random positive integer between 100 and 300.

Reply:
Yeah, I added it for the test, so we don't have to wait and can check that refresh is happening after the time the `_get_refresh_sleep_time` function returns.

Reply:
Why not det it to eg 0?

Reply:
Yes, 0 works as well and might be a better option after all 👍🏻.